internal/pkg/cli/task_run.go

```diff
@@ -376,13 +379,25 @@ func (o *runTaskOpts) Execute() error {
 		return err
 	}
 
+	if o.env == "" {
+		hasDefaultCluster, err := o.defaultClusterGetter.HasDefaultCluster()
+		if err != nil {
+			return err
```

Comment:
should we wrap this error? maybe the same way as what you have below I like the err message:
```suggestion
			return fmt.Errorf(`find "default" cluster to deploy the task to: %v`, err)
```